### PR TITLE
[Form] Fixed compatibility with FormTypeInterface implementations that don't extend AbstractType

### PR DESCRIPTION
**Pull request by webmozart**
| Q | A |
| --- | --- |
| Bug fix? | yes |
| New feature? | no |
| BC breaks? | no |
| Deprecations? | no |
| Tests pass? | yes |
| Fixed tickets | - |
| License | MIT |
| Doc PR | - |

### CODE REVIEW

File: src/Symfony/Component/Form/ResolvedFormType.php [hunk 1]
```diff
@@ -200,27 +200,36 @@ public function getOptionsResolver()
...
-            if ($isOldOverwritten && !$isNewOverwritten) {
-                @trigger_error(get_class($this->innerType).': The FormTypeInterface::setDefaultOptions() method is deprecated since version 2.7 and will be removed in 3.0. Use configureOptions() instead. This method will be added to the FormTypeInterface with Symfony 3.0.', E_USER_DEPRECATED);
+                if ($isOldOverwritten && !$isNewOverwritten) {
+                    @trigger_error(get_class($this->innerType).': The FormTypeInterface::setDefaultOptions() method is deprecated since version 2.7 and will be removed in 3.0. Use configureOptions() instead. This method will be added to the FormTypeInterface with Symfony 3.0.', E_USER_DEPRECATED);
+                }
+            } else {
+                @trigger_error(get_class($this->innerType).': The FormTypeInterface::configureOptions() method will be added in Symfony 3.0. You should extend AbstractType or implement it in your classes.', E_USER_DEPRECATED);
             }
 
+
```
Comment (stof) [inline src/Symfony/Component/Form/ResolvedFormType.php]: this extra line should be removed (see fabbot)

**Comment by stof** [comment]
:+1:

**Comment by xabbuh** [comment]
:+1:

**Comment by fabpot** [comment]
Thank you @webmozart.

**Pull request merged**